pkg/controller/messaginginfra/cert/controller.go

```diff
@@ -209,20 +391,40 @@ func (c *CertController) applyCertSecret(secret *corev1.Secret, caSecret *corev1
 		if expiryDate.After(caExpiryDate) {
 			expiryDate = caExpiryDate
 		}
-		key, cert, keystore, truststore, err := generateCert(secret.Data["tls.key"], secret.Data["tls.crt"], caSecret.Data["tls.key"], caSecret.Data["tls.crt"], expiryDate, commonName, dnsNames)
+		key, cert, keystore, truststore, err := generateCert(now, secret.Data[config.key], secret.Data[config.crt], caSecret.Data["tls.key"], caSecret.Data["tls.crt"], expiryDate, commonName, dnsNames)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		secret.Data = make(map[string][]byte)
-		secret.Data["tls.key"] = key
-		secret.Data["tls.crt"] = cert
-		secret.Data["ca.crt"] = caSecret.Data["tls.crt"]
-		secret.Data["keystore.p12"] = keystore
-		secret.Data["truststore.p12"] = truststore
+		if config.key != "" {
+			secret.Data[config.key] = key
+		}
```

Comment:
I know we don't expect the shape of the secret to change, but we avoid the risk of leaving stale values behind if we do this:
```suggestion
		} else {
			   remove(secret.Data[config.key])
		}
```